src/compat.h

```diff
@@ -16,9 +16,24 @@
 #include <pgstat.h>
 #include <utils/lsyscache.h>
 #include <utils/rel.h>
+#include <access/xact.h>
 
 #include "export.h"
 
+/* This is a copy of PreventCommandIfReadOnly() where we only do the function
+ * call if the transaction is a read-only. If we were to use the function
+ * directly, we would evaluate the function name regardless of whether we were
+ * in a read-only transaction or not */
+#define TS_PREVENT_FUNC_IF_READ_ONLY()                                                             \
+	do                                                                                             \
+	{                                                                                              \
+		if (XactReadOnly)                                                                          \
```

Comment:
Not sure if that is a good idea to have own implementation, what if the implementation changes in future versions? Yet I understand that it avoid from having memory allocation. Anyway, I fine to keep it

Reply:
Yes, that is a problem, but since the implementation is so small, I think it is fine. Some of the other `Prevent..` functions are more complicated, so that is a different story.

Reply:
I would prefer to see the postgres implementation being used. Your suggested solution breaks localization.

Reply:
The macro does not seem to play well with Windows, so reverted to use the function call.